components/authn-service/tokens/pg/adapter.go

```diff
@@ -72,24 +72,30 @@ func (a *adapter) ResetToV1(ctx context.Context) error {
 }
 
 func (a *adapter) insertToken(ctx context.Context,
-	id string, description string, value string, active bool, projects []string) (*tokens.Token, error) {
+	id string, description string, value string, active bool, projects []string, validateProjects bool) (*tokens.Token, error) {
 
 	t := tokens.Token{}
 	// ensure we do not pass null projects to db and break the not null constraint
 	if projects == nil {
 		projects = []string{}
 	}
-	_, err := a.validator.ValidateProjectAssignment(ctx, &authz_v2.ValidateProjectAssignmentReq{
-		Subjects:        auth_context.FromContext(auth_context.FromIncomingMetadata(ctx)).Subjects,
-		OldProjects:     []string{},
-		NewProjects:     projects,
-		IsUpdateRequest: false,
-	})
-	if err != nil {
-		return nil, err
+
+	// we do not wish to validate projects for the A1 legacy token case
+	// since that is an internal DB call kicked off as part of the server startup code
+	// with no subjects
+	if validateProjects {
```

Comment:
I think since we are only skipping validation for this single use, we could name this bool `isLegacyToken` or something to indicate how isolated this case is.